src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -3278,12 +3281,121 @@ namespace OpenLoco::Vehicles
         return false;
     }
 
+    // 0x0047DFD0
+    static void sub_47DFD0(VehicleHead& head, World::Pos3 pos, Track::TrackConnections& connections, bool unk)
+    {
+        // ROAD only
+        static loco_global<World::Track::TrackConnections, 0x0113609C> _113609C;
+        _113609C = connections;
+
+        registers regs;
+        regs.ax = pos.x;
+        regs.cx = pos.y;
+        regs.dx = pos.z | (unk ? 0x8000 : 0);
+        regs.esi = X86Pointer(&head);
+        call(0x0047DFD0, regs);
+    }
+
+    // 0x004AC3D3
+    static void sub_4AC3D3(VehicleHead& head, World::Pos3 pos, Track::TrackConnections& connections, bool unk)
+    {
+        // TRACK only
+        static loco_global<World::Track::TrackConnections, 0x0113609C> _113609C;
+        _113609C = connections;
+
+        registers regs;
+        regs.ax = pos.x;
+        regs.cx = pos.y;
+        regs.dx = pos.z | (unk ? 0x8000 : 0);
+        regs.esi = X86Pointer(&head);
+        call(0x004AC3D3, regs);
+    }
+
     // 0x004ACCDC
     bool VehicleHead::sub_4ACCDC()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        return call(0x004ACCDC, regs) & X86_FLAG_CARRY;
+        auto train = Vehicle(head);
+        if (mode == TransportMode::rail)
```

Comment:
These are essentially two specialisations. Would be nicer to split this into two functions, I think.